src/factory/src/state.rs

```diff
@@ -62,35 +62,36 @@ impl State {
 struct StorableWasm(Option<Vec<u8>>);
 
 impl Storable for StorableWasm {
-    fn to_bytes(&self) -> std::borrow::Cow<[u8]> {
+    fn to_bytes(&self) -> std::borrow::Cow<'_, [u8]> {
         Encode!(self)
             .expect("failed to encode StorableWasm for stable storage")
             .into()
     }
 
-    fn from_bytes(bytes: Vec<u8>) -> Self {
+    fn from_bytes(bytes: Cow<'_, [u8]>) -> Self {
         Decode!(&bytes, Self).expect("failed to decode StorableWasm from stable storage")
     }
 }
 
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
 struct StringKey(String);
 
 impl Storable for StringKey {
     fn to_bytes(&self) -> std::borrow::Cow<'_, [u8]> {
```

Comment:
```suggestion
    fn to_bytes(&self) -> Cow<'_, [u8]> {
```
Since you have imported this, why not use this ?

Reply:
Fixed